UV support to enable virtual environment and dependency version management

EVWorth opened the pull request:
![image](https://github.com/user-attachments/assets/cd164278-5d6f-42be-bf4e-3a4c84c00621)

[https://github.com/astral-sh/uv](url)

Replaces the requirements_test.txt file with pyproject.toml and uv.lock
updated tests README.md to leverage UV for running pytest.
All updated commands executed correctly with the same behavior as runing pytest directly.
![image](https://github.com/user-attachments/assets/36e79c37-e84c-46ee-bace-006873111843)

EVWorth commented:
would close #1189

cayossarian commented:
In general it looks good.  Thanks for taking the time to update the dependencies too.

HA python dependency is currently >=3.13.2 
For tests we want the latest version of home assistant to catch changes so perhaps remove that version entirely

EVWorth commented:
I'll try to take a look at this over the weekend! Thanks!

EVWorth commented:
@nielsfaber @cayossarian 

- dropped the requires-python version from 3.13.3 -> 3.13.2
-  updated the pyproject.toml to unlock the HA version to allow for testing against the latest version of Home Assistant. 
-  added a DEVELOPMENT.md file with setup docs and updated the test docs to handle updating the .venv 
-  Updated the README.md file to reference making contributions (and link to DEVELOPMENT.md)
